tests/test_ibrain.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+from gym import spaces
+
+from brains.ffnn import FeedForwardNumPy, FeedForwardPyTorch
+
+
+class TestIBrain:
+    low = -5.0
+    high = 5.0
+    one_dimensional_shape = (20,)
+    multi_dimensional_shape = (30, 15)
+    input_n = 50
+    output_n = 10
+
+    def test_size_from_space(self):
+        test_discrete = spaces.Discrete(self.input_n)
+        assert FeedForwardNumPy._size_from_space(test_discrete) == self.input_n
+
+        test_box = spaces.Box(self.low, self.high, shape=self.multi_dimensional_shape)
+        assert FeedForwardNumPy._size_from_space(test_box) == np.prod(self.multi_dimensional_shape)
+
+        test_tuple = spaces.Tuple([test_discrete, test_box])
+        assert FeedForwardNumPy._size_from_space(test_tuple) == (self.input_n + np.prod(self.multi_dimensional_shape))
+
+    # @pytest.mark.skip(reason="Not yet implemented")
+    def test_parse_box_space(self, ffnn_config):
+        input_space = spaces.Box(low=self.low, high=self.high, shape=self.one_dimensional_shape)
+        output_space = spaces.Box(low=self.low, high=self.high, shape=self.multi_dimensional_shape)
+
+        # First test the input space
+
+        # Not used but needed to create a Brain
+        dummy_individual = np.zeros((FeedForwardNumPy.get_individual_size(
+            ffnn_config, input_space=input_space, output_space=output_space)))
+
+        brain = FeedForwardPyTorch(input_space=input_space, output_space=output_space, individual=dummy_individual,
+                                   config=ffnn_config)
+
+        sample_input = input_space.sample()
+        parsed_input = brain.parse_brain_input_output(sample_input, is_brain_input=True)
+
+        # When input_space == spaces.Box the input is not transformed
+        assert np.array_equal(sample_input, parsed_input)
+
+        # Environments with RGB observations usually use spaces.Box as their observation space, which means it is
+        # the input space for the brains
+        # Assume now we have RGB input
+        sample_input = input_space.sample().reshape((1, 1, -1))
+        parsed_input = brain.parse_brain_input_output(sample_input, is_brain_input=True)
+
+        # When input_space == spaces.Box the input is not transformed
+        assert np.array_equal(sample_input, parsed_input)
+        assert isinstance(parsed_input, np.ndarray) and len(parsed_input.shape) == 3
+
+        # Now test the output space
+
+        # Reference shape, this is what the output of the brain should look like
+        sample_output_shape = output_space.sample().shape
```

Comment:
iirc you can access the shape of a space directly

```suggestion
        sample_output_shape = output_space.shape
```